pinot-core/src/main/java/com/linkedin/pinot/core/segment/index/ColumnMetadata.java

```diff
@@ -157,9 +164,27 @@ public static ColumnMetadata fromPropertiesConfiguration(String column, Properti
       }
     }
 
+
+    String partitionFunctionString = config.getString(getKeyFor(column, V1Constants.MetadataKeys.Column.PARTITION_FUNCTION));
```

Comment:
If you think of it as an attribute of the column that indicates values of this column are expected to follow a certain mathematical constraint, then it seems more appropriate to be property of a column. Also, are there any existing segment level properties with column as key (I could not find one using example segments).